### PR TITLE
 Bump github.com/gardener/gardener to 1.86.0

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area open-source
/kind enhancement
**What this PR does / why we need it**:
This PR updates the `github.com/gardener/gardener` dependency to `v1.86.0`. As a consequence, admission CA and service certificates are now managed automatically. The vendor directory is removed as well.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Bump github.com/gardener/gardener to 1.86.0.
```
```breaking operator
CA and server certificates for the admission component are managed automatically. Passing custom certificates via Helm values is not supported anymore.
```
```other developer
The `vendor` directory was removed in favor of the `go mod cache`.
```

### CODE REVIEW

File: cmd/gardener-extension-shoot-rsyslog-relp-admission/app/app.go [hunk 1]
```diff
@@ -104,13 +102,6 @@ func NewAdmissionCommand(ctx context.Context) *cobra.Command {
 					return err
 				}
 				managerOptions.LeaderElectionConfig = sourceClusterConfig
-			} else {
-				// Restrict the cache for secrets to the configured namespace to avoid the need for cluster-wide list/watch permissions.
-				managerOptions.Cache = cache.Options{
-					ByObject: map[client.Object]cache.ByObject{
-						&corev1.Secret{}: {Namespaces: map[string]cache.Config{webhookOptions.Server.Completed().Namespace: {}}},
-					},
-				}
```
Comment (ialidzhikov) [inline cmd/gardener-extension-shoot-rsyslog-relp-admission/app/app.go]: @plkokanov we should consider using APIReader to read the Project Secret in the admission component: https://github.com/gardener/gardener-extension-registry-cache/blob/3b520080f3e608720ab1c3bf1aeaa24ac31886ff/pkg/admission/validator/cache/shoot.go#L124-L128

Anyways, this is out of scope of @timuthy 's PR.

**Review by ialidzhikov — COMMENTED** [review]
Minor nit, otherwise lgtm.

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few more nits that I saw.

**Comment by plkokanov** [comment]
🚀 
/lgtm
/approve

**Pull request merged**